src/services/refactors/extractMethod.ts

```diff
@@ -772,116 +762,140 @@ namespace ts.refactor.extractMethod {
             changeTracker.replaceNodeWithNodes(context.file, range.range, newNodes, { nodeSeparator: context.newLineCharacter });
         }
 
-        return {
-            scope,
-            scopeDescription: getDescriptionForScope(scope),
-            changes: changeTracker.getChanges()
-        };
+        const edits = changeTracker.getChanges();
+        const renameRange = isReadonlyArray(range.range) ? range.range[0] : range.range;
 
-        function getFirstDeclaration(type: Type): Declaration | undefined {
-            let firstDeclaration = undefined;
+        const renameFilename = renameRange.getSourceFile().fileName;
+        const renameLocation = getRenameLocation(edits, renameFilename, functionNameText);
+        return { renameFilename, renameLocation, edits };
+    }
 
-            const symbol = type.symbol;
-            if (symbol && symbol.declarations) {
-                for (const declaration of symbol.declarations) {
-                    if (firstDeclaration === undefined || declaration.pos < firstDeclaration.pos) {
-                        firstDeclaration = declaration;
-                    }
+    function getRenameLocation(edits: ReadonlyArray<FileTextChanges>, renameFilename: string, functionNameText: string): number {
+        let delta = 0;
+        for (const { fileName, textChanges } of edits) {
+            Debug.assert(fileName === renameFilename);
+            for (const change of textChanges) {
+                const { span, newText } = change;
+                const index = newText.indexOf(functionNameText);

```

Comment:
we do not know if the name is really unique text wise. we only look if it was declared as an identifier, but not in a comment or string literal for instance.

Reply:
this also relies on the use site being inserted in the file before the definition. we should add a comment to that meaning in case we change where the function declaration is added.

Reply:
It would definitely be preferable not to assume that the new declaration is inserted after the extracted range because that will almost certainly not be the case when we start extracting locals.